inference-engine/thirdparty/movidius/XLink/shared/include/XLink.h

```diff
@@ -180,6 +188,33 @@ XLinkError_t XLinkReleaseData(streamId_t streamId);
  */
 XLinkError_t XLinkGetFillLevel(streamId_t streamId, int isRemote, int* fillLevel);
 
+/**
+ * @brief Reads data from local stream with timeout in ms. Will only have something if it was written to by the remote.
+ * Limitations.
+ *      If we reached timeout and the event waiter returned timeout error code
+ *      there potentially will be a time frame when XLink still has not marked the read
+ *      event as completed with timeout state but the event receiver has just received data
+ *      from another end and match that data with the read event. In this case we can lose
+ *      data. Should be fixed for the next release.
+ * @param[in]   streamId – stream link Id obtained from XLinkOpenStream call
+ * @param[out]  packet – structure containing output data buffer and received size
+ * @param[in]   timeoutMs – timeout for a read operation in milliseconds
+ * @return Status code of the operation: X_LINK_SUCCESS (0) for success
+ */
+XLinkError_t XLinkReadDataWithTimeout(streamId_t streamId, streamPacketDesc_t** packet, unsigned int timeoutMs);
+
+/**
+ * @brief Sends a package to initiate the writing of data to a remote stream with timeout in ms
+ * XLinkWriteDataWithTimeout is not fully supported yet. The XLinkWriteData method is called instead.
+ * @warning Actual size of the written data is ALIGN_UP(size, 64)
+ * @param[in] streamId – stream link Id obtained from XLinkOpenStream call
+ * @param[in] buffer – data buffer to be transmitted
+ * @param[in] size – size of the data to be transmitted
+ * @param[in] timeoutMs – timeout for a write operation in milliseconds
+ * @return Status code of the operation: X_LINK_SUCCESS (0) for success
+ */
+XLinkError_t XLinkWriteDataWithTimeout(streamId_t streamId, const uint8_t* buffer, int size, unsigned int timeoutMs);
```

Comment:
Looking forward to this being able to be used to gracefully exit a thread that is blocked on this operation, waiting for an incoming/outgoing packet.

Also from my naive testing of adding an equivalent function (by adding just an event with timeout), I had bunch of issues - like events would be left in dispatcher. 

Would this upcoming changes allow for "polling" of packets with specific timeout in a tight loop without any difference compared to a blocking call?